### PR TITLE
Small additions

### PR DESCRIPTION
**Pull request by pma-pj**
Hi, added/adjusted some stuff:

- sorted roles, by name (A-Z)
- searchbar, role name filtering
- centered infolabel

### CODE REVIEW

File: lua/autorun/client/cl_RoleVote_vgui_list.lua [hunk 1]
```diff
@@ -65,17 +66,34 @@ function PANEL:Init()
...
+		infoLabel:SetText("Vote for a role that will be activated until the next vote")
 	end
-
 	infoLabel:SetFont(self.font)
 	infoLabel:DockMargin(0, 0, 0, 5)
 	infoLabel:Dock(TOP)
+	infoLabel:SetContentAlignment(5) -- center
+
+	-- add a search bar
+	local searchBar = vgui.Create("DSearchBarTTT2", self)
```
Comment (Blaubeeree) [inline lua/autorun/client/cl_RoleVote_vgui_list.lua:82]:
```suggestion
	local searchBar = vgui.Create("DSearchBarTTT2", self)
	searchBar:SetUpdateOnType(true)
```
Update Filter while typing

File: lua/autorun/client/cl_RoleVote_vgui_list.lua [hunk 2]
```diff
@@ -85,6 +103,25 @@ function PANEL:Init()
...
 	buttonContainer:Dock(FILL)
 
+	-- searchbar value change listener
+	function searchBar:OnValueChange(val)
+		-- check if val is empty
+		if val ~= "" then
+			-- filter buttons for val match
+			for _, child in ipairs(buttonContainer:GetChildren()) do
+				-- set visible to val find in name
+				child:SetVisible(string.find(child:GetName(), val))
```
Comment (Blaubeeree) [inline lua/autorun/client/cl_RoleVote_vgui_list.lua]:
```suggestion
				local displayName = LANG.TryTranslation(child:GetName()):lower()
				child:SetVisible(displayName:find(val:lower()))
```
Search for the in the UI displayed name, not the internal name and ignore case

File: lua/autorun/server/sv_RoleVote.lua [hunk 3]
```diff
@@ -76,7 +76,7 @@ function RoleVote:Start(time)
 	if not RoleVote.started then
 		votes = {}
 		RoleVote.started = true
-		time = time and time or 20
+		time = time and time or GetConVar("ttt_rolevote_votetime"):GetInt()
```
Comment (Blaubeeree) [inline lua/autorun/server/sv_RoleVote.lua]: The vote should start after mapchange with infinite time and automatically end when the first round begins, if it does not thats a bug but its working for me. So the convar seems useless to me.
You can see this with the countdown in the title of the RoleVote window, thats being displayed when using `ulx rv_start` but missing after mapchange.

image

**Review by Blaubeeree — CHANGES_REQUESTED** [review]

File: lua/autorun/server/sv_RoleVote.lua (hunk 3 above)
Comment (pma-pj) [reply lua/autorun/server/sv_RoleVote.lua]: Yep doesn't make any sense..
Removed the commits.

**Review by Blaubeeree — APPROVED** [review]

**Pull request merged**